### PR TITLE
libgphoto2: conan v2 support + allow MinGW build

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
v2 pipeline needs https://github.com/conan-io/conan-center-index/pull/16099

**Comment by SpaceIm** [comment]
draft until https://github.com/conan-io/conan-center-index/pull/16099 is merged

File: recipes/libgphoto2/all/conanfile.py [hunk 1]
```diff
@@ -13,122 +19,113 @@ class LibGphoto2(ConanFile):
...
-        self._autotools.libs = []
-        yes_no = lambda v: "yes" if v else "no"
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+        if not cross_building(self):
+            env = VirtualRunEnv(self)
+            env.generate(scope="build")
```
Comment (prince-chrismc) [inline recipes/libgphoto2/all/conanfile.py:86]: I dont see any requires in the build scope can this be removed?

File: recipes/libgphoto2/all/conanfile.py [hunk 2]
```diff
@@ -13,122 +19,113 @@ class LibGphoto2(ConanFile):
...
+            self.requires("libxml2/2.10.3")
         if self.options.with_libexif:
             self.requires("libexif/0.6.23")
         if self.options.with_libjpeg:
-            self.requires("libjpeg/9d")
+            self.requires("libjpeg/9e")
+
+    def validate(self):
+        if is_msvc(self):
+            raise ConanInvalidConfiguration("Visual Studio not supported yet")
```
Comment (prince-chrismc) [inline recipes/libgphoto2/all/conanfile.py:68]: Lets encourage them to contribute if they see this message :)

```suggestion
            raise ConanInvalidConfiguration(f"{self.ref} currently Visual Studio is not supported. Contributions are welcomed.")
```

**Review by prince-chrismc — APPROVED** [review]

File: recipes/libgphoto2/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/libgphoto2/all/conanfile.py:86]: no, see autotools template for an explanation

**Review by SSE4 — APPROVED** [review]

**Pull request merged**